src/util/measureBundle.js

```diff
@@ -0,0 +1,111 @@
+const { ServerError } = require('@asymmetrik/node-fhir-server-core');
+const _ = require('lodash');
+const { findResourceById, findOneResourceWithQuery } = require('../util/mongo.controller');
+
+function mapResourcesToCollectionBundle(resources) {
+  return {
+    resourceType: 'Bundle',
+    type: 'collection',
+    entry: resources.map(r => ({
+      resource: r
+    }))
+  };
+}
+
+function isValidLibraryUrl(s) {
+  const urlRegex = /(ftp|http|https):\/\/(\w+:{0,1}\w*@)?(\S+)(:[0-9]+)?(\/|\/([\w#!:.?+=&%@!\-/]))?/;
+  return urlRegex.test(s);
+}
+
+/**
+ * Assemble a measure bundle with necessary FHIR Library resources
+ * @param {string} measureId id of the measure to assemble bundle for
+ * @returns FHIR Bundle of Measure resource and all dependent FHIR Library resources
+ */
+async function getMeasureBundleFromId(measureId) {
+  const measure = await findResourceById(measureId, 'Measure');
+
+  if (!measure) {
+    throw new ServerError(null, {
+      statusCode: 400,
+      issue: [
+        {
+          severity: 'error',
+          code: 'internal',
+          details: {
+            text: `Measure with id ${measureId} does not exist in the server`
+          }
+        }
+      ]
+    });
+  }
+
+  const [mainLibraryRef] = measure.library;
+
+  // Library references could be canonical or resourceType/id
+  let mainLib;
+  if (isValidLibraryUrl(mainLibraryRef)) {
+    mainLib = await findOneResourceWithQuery({ url: mainLibraryRef }, 'Library');
```

Comment:
This doesn't handle the situation where the canonical url includes the version.

Reply:
Good catch, added a function to support this